src/makeCalculator.js

```diff
@@ -4,7 +4,49 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const calculate = {
+    result: 0,
+
+    add(number) {
+      this.result += number;
+    },
+
+    subtract(number) {
+      this.result -= number;
+    },
+
+    multiply(number) {
+      this.result *= number;
+    },
+
+    divide(number) {
+      if (number === 0) {
+        console.log('Cannot divide by zero');
+      }
+
+      this.result /= number;
+    },
+
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+
+    operate(callback, number) {
+      const isCalculatorMethod = [this.add, this.subtract, this.multiply, this.divide].includes(callback);
+
+      if (isCalculatorMethod && typeof callback === 'function') {
+        callback.call(this, number);
+
+        return this;
+      }
+
+      return this;
```

Comment:
The `return this;` statement is duplicated in both the `if` block and outside of it. You can simplify the code by having a single return statement at the end of the `operate` method.